composer/algorithms/seq_length_warmup/seq_length_warmup.py

```diff
@@ -171,6 +237,103 @@ def __init__(
+
+        try:
+            # Both PyTorch and FFCV dataloaders define a `batch_size` attribute
+            # This exception would mainly be raised if the user is passing in a custom
+            # iterable
+            per_gpu_macrobatch = getattr(state.dataloader, 'batch_size')
+        except AttributeError as e:
+            raise AttributeError(
+                'Sequence Length Warmup requires the `state.dataloader` to have a `batch_size` attribute.') from e
+        if per_gpu_macrobatch is None:
+            raise RuntimeError('Sequence Length Warmup algorithm requires constant batch size.')
+
+        # truncate all sequence-shaped tensors to the max sequence length
+        batch_clone = {k: torch.clone(v) for k, v in state.batch.items()}
+        device_batch_size = 0
+        device = None
+        for k, v in batch_clone.items():
+            if len(v.shape) > 1:
+                batch_clone[k] = v[:, :self.max_seq_length].contiguous()
+            device_batch_size = v.shape[0]
+            device = v.device
+
+        # This loop tries to do a forward/backward pass using the current microbatch size.
+        # If it hits an OOM error, it doubles `state.grad_accum` and tries again until
+        # it succeeds.
+        grad_accum_successful = False
+        while not grad_accum_successful:
+            per_gpu_batch = ceil(per_gpu_macrobatch / state.grad_accum)
+            model_inputs = {k: v[:per_gpu_batch] for k, v in batch_clone.items()}
+
+            should_handle_cuda_oom = 0
+            caught_timeout_error = None
+            try:
+                # start by running a forward and backward pass
+                # of the maximum sequence length to allocate cache.
+                with get_precision_context(state.precision):
+                    outputs = state.model.forward(model_inputs)
+                    loss = self._original_model.loss(outputs, model_inputs)
+
+                # since use_grad_scaling is in the Trainer, and we
+                # don't care about the loss values, skip scaling
+                for loss_item in ensure_tuple(loss):
+                    loss_item.backward()
+
+                # Zero any gradients created by the backward pass
+                for optimizer in state.optimizers:
+                    optimizer.zero_grad()
+
+            # This error/state.grad_accum handling mimics the logic in trainer._train_batch().
+            except RuntimeError as e:
+                if 'CUDA out of memory' in str(e):
+                    should_handle_cuda_oom = 1
+                elif 'Timed out' in str(e):
+                    # Catch timeout errors and only reraise if we did not encounter OOM on other ranks. Error
+                    # is likely transient if one rank OOMed, it likely did not reach a barrier. Note that if we
+                    # catch non-transient timeout errors they will be later reraised if no rank OOMed.
+                    caught_timeout_error = e
```

Comment:
This can be deleted. It's for handling CUDA OOMs with gradient syncing, but the current design is flakey and we'll rework it. In either case, there's no optimizer step / gradient syncing here anyways so it's safe to remove